### ISSUE TITLE
Compile error when patching

### ISSUE THREAD

**Issue by Snowflake8** [p-web-git/hostapd#1]
When rebuilding after patching I got a compile error because of this line:
`if (iface->conf->hw_mode_set && iface->conf->hw_mode == HOSTAPD_MODE_IEEE80211A)`
saying that `hw_mode_set` does not exist and suggesting it should be `hw_mode` instead.

I'm on hostapd/stable version 2:2.10-12+deb12u2 (on bookworm)

(Thank you for the patch btw)

**Comment by PedroFirmware** [comment]
Hello

Which version of hostapd are you trying to compile? 
I have created this new patch for version [2_10](https://github.com/p-web-git/hostapd/blob/v2_10/For_intel_Wi-Fi_cards_v2_10.patch). I don't have my computer now to test it, so let me know if works please

**Comment by Snowflake8** [comment]
That's the thing, I'm doing it on version [2:2.10-12+deb12u2](http://ftp.debian.org/debian/pool/main/w/wpa/hostapd_2.10-12+deb12u2_amd64.deb), which is a 2_10 version afaict. 

One I replaced the `hw_mode_set` by `hw_mode` in the [line](https://github.com/p-web-git/hostapd/blob/e38d3c28d2590c07ede5093886e970fe31547fb6/For_intel_Wi-Fi_cards_v2_10.patch#L83) of the patch, it compiled fine.

The only issue I could see is that rerunning hostapd with the same conf file made it fail until I ran another conf, I'm guessing this has something to do with the flow control (never reaching the HT_SCAN phase), but I can live with this. I was simply wondering if the compile error is indicative of anything else or if the change could lead to any adverse effects.

To show what I mean :
![IMG_2555](https://github.com/user-attachments/assets/d5ad5a43-f0be-4745-89a1-c487361a5d16)

**Comment by PedroFirmware** [comment]
I had created the patch for the version 2.11 but when creating the name I wrote 2.10. I have push both patches to master so it should all be good now. Please download again the version 2.10 of the patch and compile it again, as the change you suggest isn't exactly the same that I am suggesting.

Which commands are you using to start and stop hostapd? I am asking to see if I can replicate on my computer

**Comment by Snowflake8** [comment]
Not able to test immediately, but I'll try to get to it asap.

Not surprised your changes might be different, that was just the compiler suggestion (which I blindly followed just to see if it worked). Not sure what you mean commands, I mean I'm just running `hostapd /etc/hostapd/hostapd.conf` (with aim to later run it in a script), and then crtl+C to stop it (maybe it's this that's breaking it?).

My hostapd.conf
```
interface=wls3f3
driver=nl80211
ssid=test
channel=149
country_code=CH

hw_mode=a
ieee80211n=1
require_ht=1
ieee80211ac=1
ieee80211ax=1
require_vht=1
ieee80211d=1
ieee80211h=1
ieee80211w=2

logger_syslog=-1
logger_syslog_level=2
logger_stdout=-1
logger_stdout_level=2

# iw list band 1 capabilities
ht_capab=[LDPC][HT40+][HT40-][SHORT-GI-20][SHORT-GI-40][TX-STBC][RX-STBC1][MAX-AMSDU-7935][DSSS_CKK-40]
vht_capab=[MAX-MPDU-11454][VHT-160][RXLDPC][SHORT-GI-80][SHORT-GI-160][TX-STBC-2BY1][SU-BEAMFORMEE][MU-BEAMFORMEE]

macaddr_acl=0
wpa=2
wpa_key_mgmt=WPA-PSK
wpa_passphrase=testtest
#wpa_pairwise=TKIP
rsn_pairwise=CCMP
auth_algs=1

vht_oper_chwidth=1
vht_oper_centr_freq_seg0_idx=155
he_su_beamformer=1
he_su_beamformee=1

wmm_enabled=1
```

**Comment by PedroFirmware** [comment]
My suggestion is for you to use [systemctl start and systemctl stop](https://www.digitalocean.com/community/tutorials/how-to-use-systemctl-to-manage-systemd-services-and-units) to start and stop as it closes the client properly and also runs on the background

**Comment by Snowflake8** [comment]
Yes, I was doing this for testing purposes, trying to optimize the configuration for performance (I wanted to see any errors asap)

**Comment by PedroFirmware** [comment]
I am closing this issue as it is resolved. If anything else comes up, feel free to open a new issue
